web-app/HeimScripts/heatmap/run.R

```diff
@@ -95,6 +154,40 @@ addStats <- function(df, sorting) {
   return(df)
 }
 
+# Coefficient of variation
+coVar     <- function(x) ( sd(x)/mean(x) )
+
+# Specific implementation of range.
+normRange <- function(x) {
+  x <- removeOutliers(x)
+  max(x) - min(x)
+}
+
+# We define outliers as measurements falling outside .25 or .75
+# quantiles by more than 1.5 of interquantile range.
+removeOutliers <- function(x) {
+  qnt    <- quantile(x, probs=c(.25, .75), na.rm = TRUE)
+  H      <- 1.5 * IQR(x, na.rm = TRUE)
+  result <- x[!(x < (qnt[1] - H))]  # Below .25 by more than 1.5 IQR
+  result[!(result > (qnt[2] + H))]  # Above .75 by more than 1.5 IQR
```

Comment:
This keeps NA's in - relevant for comment where removeOutliers is called

x<-c(1:10,NA)
> x[!(x<3)]
[1]  3  4  5  6  7  8  9 10 NA